cid_test.go

```diff
@@ -763,3 +776,15 @@ func TestLoggable(t *testing.T) {
 		t.Fatalf("did not get expected loggable form (got %v)", actual)
 	}
 }
+
+func TestErrInvalidCid(t *testing.T) {
```

Comment:
Can we expand this test to a table test that:
* covers all exported `cid` package functions that return value
* makes stronger assertions that include checking error message

please?

Reply:
@masih I added some more tests. Can you please take a look? It's important to note that only `Decode`, `Cast` and `Parse` are returning `ErrInvalidCid`. Should thing such as `Prefix.Sum` also return that? It seems that the errors there are prefix-related and not CID related

Reply:
My vote is that the entire API should behave consistently. So if CID is invalid in any of the exposed functions then the appropriate error should be returned.

Reply:
I'll update it then

Reply:
@masih updated - afaik all exported functions that parse CIDs will return with `ErrInvalidCid`.

Reply:
Thank you

Add tests for the following while at it?
* `cid.CidFromBytes`
* `cid.NewCidV0`
* `cid.NewCidV1`
* `cid.PrefixFromBytes`
* `cid.ExtractEncoding`

Reply:
Why `NewCidV0` and `NewCidV1`? Those don't return errors.

The remaining ones: I added, but they're mostly duplicating existing tests (which I modified to also check for `ErrInvalidCid`).

Reply:
Right you are thanks @hacdias